clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11164,12 +11164,13 @@ def err_ext_int_max_size : Error<"%select{signed|unsigned}0 _ExtInt of bit "
 def err_esimd_glob_cant_init : Error<
   "SYCL explicit SIMD does not permit private global variable to have an initializer">;
 
-def err_sycl_kernel_incorrectly_named : Error<"%0 is an invalid kernel name type">;
-def note_invalid_type_in_sycl_kernel : Note<
+def err_nullptr_t_type_in_sycl_kernel : Error<"'std::nullptr_t' is an invalid kernel name ">;
```

Comment:
Did we want this to read `... is invalid to use in a kernel name` so that it also covers the case `Kernel<foo<std::nullptr_t>>`?

Reply:
This was the decided text:"kernel_name" is an invalid kernel name, ‘std::nullptr_t’ is declared in std namespace(hard-coded)